huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_uniagent_batch_install_test.go

```diff
@@ -0,0 +1,152 @@
+package aom
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccUniAgentBatchInstall_basic(t *testing.T) {
+	rName := acceptance.RandomAccResourceName()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckAomInstallerAgentId(t)
+			acceptance.TestAccPreCheckAomUniAgentVersion(t)
+			acceptance.TestAccPreCheckAomTestMachineAccount(t)
+			acceptance.TestAccPreCheckAomTestMachinePassword(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		// This resource is a one-time action resource and there is no logic in the delete method.
+		// lintignore:AT001
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testUniAgentBatchInstall_basic(rName),
+			},
+		},
+	})
+}
+
+func testUniAgentBatchInstall_base(rName string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_availability_zones" "test" {}
+
+data "huaweicloud_compute_flavors" "test" {
+  availability_zone = try(data.huaweicloud_availability_zones.test.names[0], "NOT_FOUND")
+  performance_type  = "normal"
+  cpu_core_count    = 2
+  memory_size       = 4
+}
+
+data "huaweicloud_images_image" "test" {
+  name        = "Ubuntu 18.04 server 64bit"
+  most_recent = true
+}
+
+resource "huaweicloud_vpc" "test" {
+  name = "%[1]s"
+  cidr = "192.168.0.0/16"
+  description = "Custom VPC created by Terraform"
+}
+
+resource "huaweicloud_vpc_subnet" "test" {
+  name       = "%[1]s"
+  cidr       = "192.168.1.0/24"
+  gateway_ip = "192.168.1.1"
+  vpc_id     = huaweicloud_vpc.test.id
+  dns_list   = ["100.125.4.25", "100.125.129.199"]
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name        = "%[1]s"
+  description = "Security group for Terraform resources"
+}
+
+resource "huaweicloud_networking_secgroup_rule" "inbound_rules" {
+  for_each = {
+    tcp  = { protocol = "tcp" }
+    udp  = { protocol = "udp" }
+    icmp = { protocol = "icmp" }
+  }
+
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  protocol          = each.value.protocol
+  port_range_min    = each.value.protocol == "icmp" ? 0 : 1
+  port_range_max    = each.value.protocol == "icmp" ? 0 : 65535
```

Comment:
```suggestion
  for_each = toset(["tcp", "udp", "icmp"])

  direction         = "ingress"
  ethertype         = "IPv4"
  protocol          = each.value
  port_range_min    = each.value == "icmp" ? 0 : 1
  port_range_max    = each.value == "icmp" ? 0 : 65535
```
The key is never used and the values are only includes one sub-key.

Reply:
for_each is not support.